sdk/test-utils/recorder/src/utils/requestBodyTransform.ts

```diff
@@ -0,0 +1,132 @@
+  if (runtime === "node") {
+    if (typeof fixture !== "string") {
+      // TODO: Handle JSON fixtures here - parts of below can be refactored
+      // Not required yet since we're not handling the browser recordings
+      return fixture;
+    }
+    // Modify the request body
+    let updatedFixture = fixture;
+
+    // TODO: PUT and PATCH may also have request bodies, currently focusing only on POST - can be extended as needed
+
+    // Matching the following at this point
+    //    .post('/azuretenantid/oauth2/v2.0/token', "client-request-id=11111111-1111-1111-1111-111111111111&client_secret=azure_client_secret")
+    //    .reply(200,....
+    let matches = fixture.match(/\.post\((.*)\, (.*)\)\n\s*.reply\(/);
+    if (
+      matches?.[2] &&
+      typeof matches[2] === "string" &&
+      requestBodyTransformations.stringTransforms
+    ) {
+      let updatedBody = matches[2]; // Must be string - either normal or JSON-stringified
+      // normal string
+      for (const transformation of requestBodyTransformations.stringTransforms) {
+        updatedBody = transformation(updatedBody);
+      }
+      // TODO: Handle JSON stringified bodies - not required as of now
+
+      // Updated fixture with the new request body
+      // Example:
+      //    .post('/azuretenantid/oauth2/v2.0/token', "client-request-id=client-request-id&client_secret=azure_client_secret")
+      //    .reply(200,....
+      updatedFixture = fixture.replace(matches[2], updatedBody);
+    }
+
+    if (updatedFixture === fixture) {
+      // No need to update the fixture with filtering method since the body didn't change
+      return fixture;
+    }
+    // Modify the updated fixture with `.filteringRequestBody` method to be able to match the request in playback
+    matches = updatedFixture.match(/\.post\((.*)\, (.*)\)\n\s*.reply\(/);
+    if (matches?.[0] && requestBodyTransformations.stringTransforms) {
+      for (const transformation of requestBodyTransformations.stringTransforms) {
+        // Add .filteringRequestBody method with the transformation in the recording
+        // Example: `.filteringRequestBody((body) => body.replace(/client-request-id=[^&]<star>/g, "client-request-id=client-request-id"))`
+        //
+        // Recording would look like the following
+        //  nock('https://login.microsoftonline.com:443', {"encodedQueryParams":true})
+        //   .filteringRequestBody((body) => body.replace(/client-request-id=[^&]<star>/g, "client-request-id=client-request-id"))
+        //   .post('/azuretenantid/oauth2/v2.0/token', "client-request-id=client-request-id&client_secret=azure_client_secret")
+        //   ...
+        //  ]);
+        updatedFixture = updatedFixture.replace(
+          matches[0],
+          `.filteringRequestBody(${transformation.toString()})\n  ` + matches[0]
+        );
+      }
+    }
+    return updatedFixture;
+  } else {
+    // TODO: Browser side - not needed right now since identity is not being upgraded to msal
```

Comment:
Identity is in fact being upgraded to MSAL, just that we won't have changes on the credentials used in the browser.

Reply:
Do we have a plan for the browser? We should at least investigate if it's possible in the browser.

- If it's possible: Then we should log a message here saying "this feature is not yet supported in the browser".
- If it's not possible, then we should throw in browsers, potentially skip this feature completely from the types in the browser, and let uses do `if (isNode)` to enable this feature themselves.

Reply:
It is feasible in browser, not implementing yet since we don't have tests using identity v2 with msal in the browser.